app/utils/ServerRequests.tsx

```diff
@@ -292,13 +314,269 @@ export const uploadFile = async (file: Blob, location: string ): Promise<boolean
         body: raw,
         redirect: "follow"
     };
-    
-    return fetch(machineURL+location, requestOptions as RequestInit)
-        .then((response) => response.text())
-        .then((result) => { return true })
+
+    return fetch(machineURL + location, requestOptions as RequestInit)
+        .then((response) => { return response.text() })
+        .then(async (result) => {
+            let fileID: number
+            await getFolderContent(location).then(async (response: void | FileListType) => {
+                if (response) {
+                    fileID = response.files[response.files.length - 1].fileID;
+                    let tags = await getAllSystemTags().then((tags: void | FileTagType[]) => {
+                        let file_name = response.files[response.files.length - 1].fileName.toLowerCase();
+                        let directory_path = location; // assuming directory_path is same as location
+
+                        if (!tags) {
+                            console.error("No tags found in system");
+                            return false;
+                        }
+
+                        // Assign tags based on directory path
+                        ExampleTags['Subjects'].forEach((tag) => {
```

Comment:
Why do we use the tags from the "Example-Tags.tsx" file here but anywhere else the tags we fetched Nextcloud?

Reply:
Bc we need to treat some tags differently, therefore we need to compare then. Exampletags are the same in the json in the backend